extensions/assetLibrary/src/client/Library.tsx

```diff
@@ -0,0 +1,466 @@
+  ejectRuntime: formatMessage('Eject Runtime'),
+  noComponentsInstalled: formatMessage('No packages installed'),
+  noComponentsFound: formatMessage('No packages found. Check extension configuration.'),
+  browseHeader: formatMessage('Browse'),
+  installHeader: formatMessage('Installed'),
+  libraryError: formatMessage('Library Error'),
+  importError: formatMessage('Install Error'),
+}
+
+const Library: React.FC = () => {
+  const [items, setItems] = useState<LibraryRef[]>([]);
+  const [groups, setGroups] = useState<any[]>([]);
+  const { settings, projectId, reloadProject } = useProjectApi();
+  const { setApplicationLevelError, navigateTo, confirm } = useApplicationApi();
+
+  const [ejectedRuntime, setEjectedRuntime] = useState<boolean>(false);
+  const [availableLibraries, updateAvailableLibraries] = useState<any[]>([]);
+  const [installedComponents, updateInstalledComponents] = useState<any[]>([]);
+  const [recentlyUsed, setRecentlyUsed] = useState<any[]>([]);
+  const [programmingLanguageSelection, setProgrammingLanguageSelection] = useState<string>('c#');
+  const [runtimeLanguage, setRuntimeLanguage] = useState<string>('c#');
+
+  const [selectedItem, setSelectedItem] = useState<LibraryRef>();
+  const [working, setWorking] = useState(false);
+  const [addDialogHidden, setAddDialogHidden] = useState(true);
+  const httpClient = useHttpClient();
+  const API_ROOT = '';
+
+  const programmingLanguages = [
+    {
+      key: 'c#',
+      text: 'C#',
+    },
+    {
+      key: 'js',
+      text: 'Javascript',
+    }
+  ];
+
+  const onChangeLanguage = (ev, op, idx) => {
+    setProgrammingLanguageSelection(op.key);
+    return true;
+  }
+
+  const installComponentAPI = async (projectId: string, packageName: string, version: string, isUpdating: boolean) => {
+    return httpClient.post(`${API_ROOT}/projects/${projectId}/import`, {
+      package: packageName,
+      version: version,
+      isUpdating,
+    });
+  };
+
+  const getLibraryAPI = async () => {
+    return httpClient.get(`${API_ROOT}/library`);
+  };
+
+  const getInstalledComponentsAPI = async (projectId: string) => {
+    return httpClient.get(`${API_ROOT}/projects/${projectId}/installedComponents`);
+  };
+
+  const uninstallComponentAPI = async (projectId: string, packageName: string) => {
+    return httpClient.post(`${API_ROOT}/projects/${projectId}/unimport`, {
+      package: packageName,
+    });
+  };
+
+  const isCompatible = (component) => {
+    return (component.language === programmingLanguageSelection);
+  }
+
+  /**
+   * This method will eventually be moved to the server, when we can store this list on disk via an extension mechanism.
+   * For now, we're doing it in the client, using LocalStorage.
+   * The goal is to capture a list of recently used components, and offer them up across projects
+   * @param componentList
+   */
```

Comment:
This can moved to the server now. Did you want to do it in this PR?

Reply:
yes